.github/workflows/samcli-integration-test.yaml

```diff
@@ -0,0 +1,309 @@
+        with:
+          role-to-assume: ${{ secrets.ROLE }}
+          role-session-name: samcli-integration-test
+          aws-region: ${{ secrets.REGION }}
+
+      - name: Checkout repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.9' # min. supported by SAMcli
+
+      - name: Set up SAM CLI from source
+        run: |
+          python -m pip install --upgrade pip
+          git clone https://github.com/aws/aws-sam-cli.git 
+          cd aws-sam-cli
+          git checkout $(git describe --tags `git rev-list --tags --max-count=1`)
+          git submodule update --init --recursive
+          make init
+          which samdev
+          samdev --version
+
+      - name: Run unit tests
+        working-directory: aws-sam-cli  
+        run: |
+
+          # Run the unit tests and capture output
+          make test | tee unit_test_output.txt
+
+          # Check if tests passed
+          if ! grep -q "failed=0" unit_test_output.txt; then
+            echo "Unit tests failed!"
+            exit 1
+          fi
+
+          # Extract coverage percentage
+          COVERAGE=$(grep -o "[0-9]\+%" unit_test_output.txt | head -1 | tr -d '%')
+
+          # Check if coverage is at least 90% (allowing for some variation from the ~94%)
+          if [ -z "$COVERAGE" ] || [ "$COVERAGE" -lt 90 ]; then
+            echo "Coverage is below expected threshold! Got: $COVERAGE%, Expected: ~94%"
+            exit 1
+          else
+            echo "Unit tests passed with $COVERAGE% coverage (expected ~94%)"
+          fi
+
+      - name: Build and initialize Finch
+        run: |
+          make
+          ./bin/finch vm init || ./bin/finch vm start
+          if ! ./bin/finch vm status | grep -q "Running"; then
+            echo "Error: Finch VM is not running"
+            exit 1
+          else
+            echo "Finch VM is running successfully"
+          fi
+
+      - name: Run local invoke test
+        timeout-minutes: 75
+        working-directory: aws-sam-cli
+        run: |
+
+          # Run the test, display output
+          python -m pytest tests/integration/local/invoke -k 'not Terraform' -v > test_output.txt 2>&1 || true
+          cat test_output.txt
+
+          # Create a list of expected failing tests
+          cat > expected_failures.txt << 'EOL'
+          test_invoke_with_error_during_image_build
+          test_invoke_with_timeout_set_0_TimeoutFunction
+          test_invoke_with_timeout_set_1_TimeoutFunctionWithParameter
+          test_invoke_with_timeout_set_2_TimeoutFunctionWithStringParameter
+          test_building_new_rapid_image_removes_old_rapid_images
+          test_invoke_returns_expected_results_from_git_function
+          test_invoke_returns_expected_results_from_git_function_with_parameters
+          EOL
```

Comment:
Not familiar with shellscripts so I could be wrong, wonder if this should be EOF vs EOL?